src/main/java/hello/controller/CustomerController.java

```diff
@@ -20,12 +21,26 @@
     @Autowired
     private CustomerRepository respository;
 
+    static Logger log = Logger.getLogger(CustomerController.class);
+
 
     @PostMapping
-    public ResponseEntity addCustomer(@RequestBody Customer customer) { //to turn json object in java customer
+    public ResponseEntity addCustomer(@RequestBody Customer customer){ //to turn json object in java customer
+        System.out.println(customer);
+        if ((customer.getName().isEmpty()) || (customer.getName() == null)) {
```

Comment:
You could consider abstracting this logic to, say, a validator class to decouple it from the controller? Where possible, controllers should only really be concerned with requests and responses

Reply:
Also there are libraries to perform the logic for this line; take a look at Apache commons `StringUtils#isBlank`